library/ws-widget/collection/src/lib/_services/widget-content.service.ts

```diff
@@ -482,15 +482,15 @@ export class WidgetContentService {
             if (!this.isBatchInProgress(enrolledCourseData.batch)) {
               return this.gotoTocPage(content)
             }
-        } else {
-          const data =  await this.checkForDataToFormUrl(content, enrolledCourseData)
-          return data
+            const returnData =  await this.checkForDataToFormUrl(content, enrolledCourseData)
+            return returnData
         }
+        const data =  await this.checkForDataToFormUrl(content, enrolledCourseData)
```

Comment:
why it is duplicated?

Reply:
it is under if block first set of  code is if the content enrolled  having closed batches 
and second block is for open batches